### PR TITLE
- sql-parser build.gradle, check for OS and adjust antlr output path accordingly (windows build support)

### PR DESCRIPTION
**Pull request by senchi**
- sql-parser build.gradle, check for OS and adjust antlr output path accordingly (windows build support)

### CODE REVIEW

File: sql-parser/build.gradle [hunk 1]
```diff
@@ -25,8 +27,14 @@ dependencies {
     testCompile 'junit:junit:4.11'
 }
 
-task generateWithANTLR3(type:Exec) { 
-    commandLine = ['java', '-cp',  configurations.antlr3.getAsPath(), 'org.antlr.Tool', '-o', '.', 'src/main/java/io/crate/sql/parser/Statement.g', 'src/main/java/io/crate/sql/parser/StatementBuilder.g']
+task generateWithANTLR3(type:Exec) {
+    String outputPath = '.';
+    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        logger.info("*** WINDOWS ");
```
Comment (dobe) [inline sql-parser/build.gradle]: pls remove logger

**Comment by senchi** [comment]
All sql module tests are now green on Windows, except for ArrayMapperMetaMigrationTest.testMigrateOldIndices which is fluctuating, but should have been already fixed on another branch.

**Comment by dobe** [comment]
pls rebase master

**Comment by senchi** [comment]
rebase master

**Comment by senchi** [comment]
Rebased again with "push -f"

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 2]
```diff
@@ -70,14 +73,14 @@ public void testStatementBuilder()
         printStatement("select x is distinct from y from foo where a is not distinct from b");
```
Comment (dobe) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: rewwrite the println function

File: sql/src/test/java/io/crate/operation/reference/sys/TestSysNodesExpressions.java [hunk 3]
```diff
@@ -519,4 +535,17 @@ public void testNestedBytesRefExpressionsString() throws Exception {
         assertThat(versionNumber.value(), is(new BytesRef(version.value().get(NodeVersionExpression.NUMBER).toString())));
 
     }
+    @Test
+    public void test() {
+        Sigar sigar = new Sigar();
```
Comment (dobe) [inline sql/src/test/java/io/crate/operation/reference/sys/TestSysNodesExpressions.java]: check from where this comes

**Comment by senchi** [comment]
Travis CI build faild on task :docs:itest, there were no changes however.. 
Could we retry travis or I'll try rebase again and force push..

Thanks

**Comment by senchi** [comment]
This time, GroupByAggregateTest fails. 
There is no diff with crate/master (i.e. I haven't changed this test or its parents), this suggests that this test fail should happen on crate/master as well.

I've re-tested locally (on Windows 8.1), and I'm getting fluctuating FAILS for different tests within the GroupByAggregateTest class (mostly testGroupByAggregateOnNestedColumn), but with the slightly different stack: 

```
SQLActionException{errorCode=4043, status=NOT_FOUND, stackTrace='io.crate.exceptions.ColumnUnknownException: Column details['job'] unknown
	at io.crate.analyze.relations.FullQualifedNameFieldProvider.resolveField(FullQualifedNameFieldProvider.java:125)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.resolveSubscriptSymbol(ExpressionAnalyzer.java:483)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitSubscriptExpression(ExpressionAnalyzer.java:475)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitSubscriptExpression(ExpressionAnalyzer.java:356)
	at io.crate.sql.tree.SubscriptExpression.accept(SubscriptExpression.java:45)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitFunctionCall(ExpressionAnalyzer.java:380)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitFunctionCall(ExpressionAnalyzer.java:356)
	at io.crate.sql.tree.FunctionCall.accept(FunctionCall.java:75)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.analyze.expressions.ExpressionAnalyzer.convert(ExpressionAnalyzer.java:134)
	at io.crate.analyze.relations.select.SelectAnalyzer$SelectAnalysis.toSymbol(SelectAnalyzer.java:80)
	at io.crate.analyze.relations.select.SelectAnalyzer$InnerVisitor.visitSingleColumn(SelectAnalyzer.java:103)
	at io.crate.analyze.relations.select.SelectAnalyzer$InnerVisitor.visitSingleColumn(SelectAnalyzer.java:99)
	at io.crate.sql.tree.SingleColumn.accept(SingleColumn.java:103)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.sql.tree.DefaultTraversalVisitor.visitSelect(DefaultTraversalVisitor.java:111)
	at io.crate.sql.tree.Select.accept(Select.java:56)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.analyze.relations.select.SelectAnalyzer.analyzeSelect(SelectAnalyzer.java:49)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuerySpecification(RelationAnalyzer.java:99)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuerySpecification(RelationAnalyzer.java:52)
	at io.crate.sql.tree.QuerySpecification.accept(QuerySpecification.java:116)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.analyze.SelectStatementAnalyzer.visitQuery(SelectStatementAnalyzer.java:46)
	at io.crate.analyze.SelectStatementAnalyzer.visitQuery(SelectStatementAnalyzer.java:32)
	at io.crate.sql.tree.Query.accept(Query.java:88)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.analyze.Analyzer$AnalyzerDispatcher.visitQuery(Analyzer.java:107)
	at io.crate.analyze.Analyzer$AnalyzerDispatcher.visitQuery(Analyzer.java:50)
	at io.crate.sql.tree.Query.accept(Query.java:88)
	at io.crate.sql.tree.AstVisitor.process(AstVisitor.java:31)
	at io.crate.analyze.Analyzer.analyze(Analyzer.java:44)
	at io.crate.action.sql.TransportSQLAction.getAnalysis(TransportSQLAction.java:69)
	at io.crate.action.sql.TransportSQLAction.getAnalysis(TransportSQLAction.java:49)
	at io.crate.action.sql.TransportBaseSQLAction.doExecute(TransportBaseSQLAction.java:186)
	at io.crate.action.sql.TransportBaseSQLAction.doExecute(TransportBaseSQLAction.java:74)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:75)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:55)
	at org.elasticsearch.client.node.NodeClient.execute(NodeClient.java:90)
	at io.crate.testing.SQLTransportExecutor.execute(SQLTransportExecutor.java:79)
	at io.crate.testing.SQLTransportExecutor.exec(SQLTransportExecutor.java:63)
	at io.crate.integrationtests.SQLTransportIntegrationTest.execute(SQLTransportIntegrationTest.java:102)
	at io.crate.integrationtests.GroupByAggregateTest.testGroupByAggregateOnNestedColumn(GroupByAggregateTest.java:569)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1618)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:827)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:877)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:168)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:50)
	at org.apache.lucene.util.TestRuleFieldCacheSanity$1.evaluate(TestRuleFieldCacheSanity.java:51)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:46)
	at com.carrotsearch.randomizedtesting.rules.SystemPropertiesInvariantRule$1.evaluate(SystemPropertiesInvariantRule.java:55)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:49)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:65)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:48)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:365)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:798)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:458)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:836)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$3.evaluate(RandomizedRunner.java:738)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$4.evaluate(RandomizedRunner.java:772)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:783)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:46)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:42)
	at com.carrotsearch.randomizedtesting.rules.SystemPropertiesInvariantRule$1.evaluate(SystemPropertiesInvariantRule.java:55)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:54)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:65)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:55)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:365)
	at java.lang.Thread.run(Thread.java:745)
'}
	at __randomizedtesting.SeedInfo.seed([7B36BE805A0D8F47:1D5DD90CD393921B]:0)
	at io.crate.action.sql.TransportBaseSQLAction.buildSQLActionException(TransportBaseSQLAction.java:357)
	at io.crate.action.sql.TransportBaseSQLAction.doExecute(TransportBaseSQLAction.java:190)
	at io.crate.action.sql.TransportBaseSQLAction.doExecute(TransportBaseSQLAction.java:74)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:75)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:55)
	at org.elasticsearch.client.node.NodeClient.execute(NodeClient.java:90)
	at io.crate.testing.SQLTransportExecutor.execute(SQLTransportExecutor.java:79)
	at io.crate.testing.SQLTransportExecutor.exec(SQLTransportExecutor.java:63)
	at io.crate.integrationtests.SQLTransportIntegrationTest.execute(SQLTransportIntegrationTest.java:102)
	at io.crate.integrationtests.GroupByAggregateTest.testGroupByAggregateOnNestedColumn(GroupByAggregateTest.java:569)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1618)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:827)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:877)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:168)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:50)
	at org.apache.lucene.util.TestRuleFieldCacheSanity$1.evaluate(TestRuleFieldCacheSanity.java:51)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:46)
	at com.carrotsearch.randomizedtesting.rules.SystemPropertiesInvariantRule$1.evaluate(SystemPropertiesInvariantRule.java:55)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:49)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:65)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:48)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:365)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:798)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:458)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:836)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$3.evaluate(RandomizedRunner.java:738)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$4.evaluate(RandomizedRunner.java:772)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:783)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:46)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:42)
	at com.carrotsearch.randomizedtesting.rules.SystemPropertiesInvariantRule$1.evaluate(SystemPropertiesInvariantRule.java:55)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:54)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:65)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:55)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:365)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by mfussenegger** [comment]
is this PR otherwise ready from your side?

**Comment by senchi** [comment]
This is PR contains sql and blob modules done and green!
dobe suggested we might want to pull in this milestone otherwise it'll be too much..

I'm working on sql-parser module right now where I'm getting strange "NoClassDefFoundError" so far.. also bit fluctuating but I need to research more there..

Thanks

**Comment by senchi** [comment]
PS. the stack of the errors in the sql-parser module look like this, and happen sometimes in `TestSqlParser` alone and sometimes in `TestStatementBuilder` as well..

```
org.testng.TestException: 
Expected exception io.crate.sql.parser.ParsingException but got java.lang.NoClassDefFoundError: Could not initialize class io.crate.sql.tree.ComparisonExpression$Type
	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1497)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:754)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1198)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1123)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.stop(TestNGTestClassProcessor.java:115)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:57)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:115)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:355)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.crate.sql.tree.ComparisonExpression$Type
	at io.crate.sql.parser.StatementBuilder.comparisonType(StatementBuilder.java:10924)
	at io.crate.sql.parser.StatementBuilder.comparisonExpression(StatementBuilder.java:10612)
	at io.crate.sql.parser.StatementBuilder.expr(StatementBuilder.java:7022)
	at io.crate.sql.parser.StatementBuilder.expr(StatementBuilder.java:7136)
	at io.crate.sql.parser.StatementBuilder.expr(StatementBuilder.java:7136)
```

**Comment by senchi** [comment]
PPS. All other modules' tests are OK.
There is one more issue left, that is the config.yml and logigng.yml files in the app/config directory (that are actually symlinks that point to real files in the dist directory).

Did you managed to find a solution for that otherwise I can have a look?

**Comment by mfussenegger** [comment]
merged via https://github.com/crate/crate/pull/1634  (I squashed all commits into 2 and changed the commit message a bit)

we didn't look into the config.yml/logging.yml issue yet, so feel free to dig into it and open another PR.

**Pull request closed without merging**